ImageFilterEditor/ImageFilterEditor/ImagePostViewController.swift

```diff
@@ -0,0 +1,207 @@
+            if let filteredImage = kaleidoscopeFilter.outputImage {
+                outputImage = filteredImage
+            }
+        }
+        
+        // PerspectiveTransform
+        perspectiveTransformFilter.inputImage = outputImage
+        let imageAspectRatio = outputImage.extent.width / outputImage.extent.height
+        let leftSideYOffset: CGFloat = perspectiveSlider.value > 0.5 ? 0 : (CGFloat(0.5 - perspectiveSlider.value) * 500)
+        let rightSideYOffset: CGFloat = perspectiveSlider.value < 0.5 ? 0 : (CGFloat(perspectiveSlider.value - 0.5) * 500)
+        let leftSideXOffset: CGFloat = imageAspectRatio * leftSideYOffset * 2
+        let rightSideXOffset: CGFloat = imageAspectRatio * rightSideYOffset * 2
+        perspectiveTransformFilter.bottomLeft = CGPoint(x: outputImage.extent.minX + leftSideXOffset,
+                                                        y: outputImage.extent.minY + leftSideYOffset)
+        perspectiveTransformFilter.bottomRight = CGPoint(x: outputImage.extent.maxX - rightSideXOffset,
+                                                         y: outputImage.extent.minY + rightSideYOffset)
+        perspectiveTransformFilter.topLeft = CGPoint(x: outputImage.extent.minX + leftSideXOffset,
+                                                     y: outputImage.extent.maxY - leftSideYOffset)
+        perspectiveTransformFilter.topRight = CGPoint(x: outputImage.extent.maxX - rightSideXOffset,
+                                                      y: outputImage.extent.maxY - rightSideYOffset)
+        if let filteredImage = perspectiveTransformFilter.outputImage {
+            outputImage = filteredImage
+        }
+        
+        // I could not figure out how to get the crystalize and pointillize filters to work
+        
+        // Crystalize
+        /*
+        if slider2.value > 0 {
+            //crystalizeFilter.inputImage = outputImage
+            crystalizeFilter.setValue(outputImage, forKey: kCIInputImageKey)
+            crystalizeFilter.radius = slider2.value * 10
+            crystalizeFilter.center = CGPoint(x: outputImage.extent.midX,
+                                              y: outputImage.extent.midY)
+            if let filteredImage = crystalizeFilter.outputImage {
+                outputImage = filteredImage
+            }
+        }
+        */
+        
+        // Pointillize
+        /*
+        if slider2.value > 0 {
+            //crystalizeFilter.inputImage = outputImage
+            pointillizeFilter.setValue(outputImage, forKey: kCIInputImageKey)
+            pointillizeFilter.radius = slider2.value * 60
+            pointillizeFilter.center = CGPoint(x: outputImage.extent.midX,
+                                               y: outputImage.extent.midY)
+            if let filteredImage = pointillizeFilter.outputImage {
+                outputImage = filteredImage
+            }
+        }
+        */
```

Comment:
I like that you left documentation for yourself but try not to ship commented code.